tools/src/commands/apply_acme_cert.rs

```diff
@@ -51,15 +40,26 @@ pub async fn main(opts: Opts) -> Result<()> {
         fetcher: Box::new(HyperFetcher::new()),
         ..Default::default()
     };
-    let challenge_answer = loop {
+    let (challenge_token, challenge_answer) = loop {
         runtime.now = std::time::SystemTime::now();
         update_acme_state_machine(&runtime, &acme_account).await?;
-        if let Some((_token, answer)) = get_challenge_token_and_answer(&runtime).await? {
-            break answer;
+        if let Some((token, answer)) = get_challenge_token_and_answer(&runtime).await? {
+            break (token, answer);
         }
         tokio::time::sleep(std::time::Duration::from_secs(1)).await;
     };
-    let tx = start_warp_server(opts.port, challenge_answer);
+    println!(
```

Comment:
Curious as to why are you removing this? Wouldn't it make it harder for users to setup?

Reply:
Using `warp` requires an extra step to close the server, but Rust does not have the `defer` keyword lie Golang. Let me revert the deletion.